engine/src/witness/sol/nonce_witnessing.rs

```diff
@@ -49,20 +52,21 @@ where
 				return Err(anyhow!("Expected nonce account, got space {:?}", space));
 			}
 
-			let info =
-				parsed.get("info").and_then(Value::as_object).ok_or(anyhow!("Info not found"))?;
+			let info = parsed
+				.get("info")
+				.and_then(Value::as_object)
+				.ok_or_else(|| anyhow!("Info not found"))?;
 			let hash = SolHash::from_str(
 				info.get("blockhash")
 					.and_then(Value::as_str)
-					.ok_or(anyhow!("Blockhash not found"))?,
+					.ok_or_else(|| anyhow!("Blockhash not found"))?,
 			)?;
 
-			result.push((*nonce_account, hash));
-		} else {
-			return Err(anyhow!("Expected UiAccountData::Json(ParsedAccount)"));
-		}
+			Ok(Some(hash))
+		},
+		Some(_) => Err(anyhow!("Expected UiAccountData::Json(ParsedAccount)")),
+		None => Ok(None),
 	}
```

Comment:
This is the main change to the nonce witnessing. 

Previously, if the rpc returned `None` we would error. Now we return `Ok(None)` to indicate that the rpc succeeded but that the account wasn't found. This should cause the vote to be filtered out, and retried later.